pkg/lib/ownerutil/util.go

```diff
@@ -123,6 +123,31 @@ func Adoptable(target Owner, owners []metav1.OwnerReference) bool {
 	return false
 }
 
+// OwnersIntersect checks to see if any member of targets exists in owners
+func OwnersIntersect(targets []Owner, owners []metav1.OwnerReference) bool {
+	if len(owners) == 0 {
+		// Resources with no owners are not adoptable
+		return false
+	}
+
+	for _, target := range targets {
+		// Infer TypeMeta for the target
+		if err := InferGroupVersionKind(target); err != nil {
```

Comment:
Do we really care about the GVK if the `name` and `uid` are the same?

Reply:
@ecordell AFAICT, some resources like `Project` and `Namespace` can share the same name and UID:
```
➜  k get ns openshift-operator-lifecycle-manager -o jsonpath={.metadata.uid}
4b309c1f-33ae-11e9-9f1e-02920433f040
➜  k get project openshift-operator-lifecycle-manager -o jsonpath={.metadata.uid}
4b309c1f-33ae-11e9-9f1e-02920433f040
```